sky/sdk/lib/widgets/animated_container.dart

```diff
@@ -75,10 +91,106 @@ class ImplicitlyAnimatedValue<T> {
   }
 }
 
+abstract class AnimationIntention {
+  void initFields(AnimatedContainer original);
+  void syncFields(AnimatedContainer original, AnimatedContainer updated);
+}
+
+abstract class ImplicitlySyncFieldIntention<T> extends AnimationIntention {
+  ImplicitlySyncFieldIntention(this.duration);
+
+  Duration duration;
+  ImplicitlyAnimatedValue<T> field;
+
+  // Overrides.
+  T getter(AnimatedContainer container);
+  void setter(AnimatedContainer container, T value);
+  AnimatedValue<T> initField(T value);
+
+  void initFields(AnimatedContainer original) {
+    _updateField(original, getter(original));
+  }
+
+  void syncFields(AnimatedContainer original, AnimatedContainer updated) {
+    _updateField(original, getter(updated));
+  }
+
+  void _updateField(AnimatedContainer original, T newValue) {
+    if (field != null) {
+      // Animate to newValue (possibly null).
+      field.animateTo(newValue);
+    } else if (newValue != null) {
+      // Set the value and prepare it for future animations.
+      field = new ImplicitlyAnimatedValue<T>(initField(newValue), duration);
+      field.performance.addListener(() {
+        original.setState(() { setter(original, field.value); });
+      });
+    }
+  }
+}
+
+class ImplicitlySyncConstraintsIntention extends ImplicitlySyncFieldIntention<BoxConstraints> {
+  ImplicitlySyncConstraintsIntention(Duration duration) : super(duration);
+
+  BoxConstraints getter(AnimatedContainer c) => c.constraints;
+  void setter(AnimatedContainer c, BoxConstraints val) { c.constraints = val; }
```

Comment:
OK. FWIW, I did it this way to save on vertical space (otherwise the lines get long and I have to wrap).